### PR TITLE
Implement SDK Functions for Cardano Pool APIs with Tests

### PR DESCRIPTION
**Pull request by hieu-dd**
# Pull Request (PR) Summary

- This PR introduces new functions in the SDK for interacting with Cardano Pool APIs, along with comprehensive unit and integration tests.

## Changes Description

This update adds a series of functions to the SDK that enable interactions with the Cardano Pool APIs. These APIs are crucial for accessing and managing data related to stake pools on the Cardano blockchain. Stake pools are essential for the operation of Cardano's proof-of-stake (PoS) mechanism, and these functions allow users to query and manage pool information effectively.

### New Functions
- Functions to retrieve information about specific pools, list all pools, and get pool statistics.
- Functions to interact with pool registration and retirement processes.

### Testing
- **Unit Tests**: Each new function is accompanied by unit tests that validate its functionality in isolation, ensuring that each component behaves as expected.
- **Integration Tests**: Integration tests are conducted to ensure that these functions work correctly when interacting with actual Cardano network endpoints and other parts of the SDK.

These tests are designed to cover various scenarios and edge cases, ensuring robustness and reliability of the SDK functions.

## Related Issue(s)

- Fixes # (issue number)  // Please replace with actual issue number if applicable

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please describe)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] My PR is targeting the `main` branch
- [x] I have added necessary documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or my feature works
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/src/commonTest/resources/model/network_information.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "supply": {
+    "max": "45000000000000000",
```
Comment (iTanChi) [inline core/src/commonTest/resources/model/network_information.json]: những thông tin ntn thì khai báo ở code là BigInteger nhé

**Review by iTanChi — APPROVED** [review]
